pkg/agent/ipassigner/responder/arp_responder.go

```diff
@@ -65,40 +57,88 @@ func (r *arpResponder) RemoveIP(ip net.IP) error {
 		return fmt.Errorf("only IPv4 is supported")
 	}
 	if r.deleteIP(ip) {
-		klog.InfoS("Removed IP from ARP responder", "ip", ip, "interface", r.iface.Name)
+		klog.InfoS("Removed IP from ARP responder", "ip", ip, "interface", r.linkName)
 	}
 	return nil
 }
 
-func (r *arpResponder) handleARPRequest() error {
-	pkt, _, err := r.conn.Read()
+func (r *arpResponder) handleARPRequest(client *arp.Client, iface *net.Interface) error {
+	pkt, _, err := client.Read()
 	if err != nil {
 		return err
 	}
 	if pkt.Operation != arp.OperationRequest {
 		return nil
 	}
 	if !r.isIPAssigned(pkt.TargetIP) {
-		klog.V(4).InfoS("Ignored ARP request", "ip", pkt.TargetIP, "interface", r.iface.Name)
+		klog.V(4).InfoS("Ignored ARP request", "ip", pkt.TargetIP, "interface", r.linkName)
 		return nil
 	}
-	if err := r.conn.Reply(pkt, r.iface.HardwareAddr, pkt.TargetIP); err != nil {
+	if err := client.Reply(pkt, iface.HardwareAddr, pkt.TargetIP); err != nil {
 		return fmt.Errorf("failed to reply ARP packet for IP %s: %v", pkt.TargetIP, err)
 	}
-	klog.V(4).InfoS("Sent ARP response", "ip", pkt.TargetIP, "interface", r.iface.Name)
+	klog.V(4).InfoS("Sent ARP response", "ip", pkt.TargetIP, "interface", r.linkName)
 	return nil
 }
 
 func (r *arpResponder) Run(stopCh <-chan struct{}) {
+	r.once.Do(func() {
```

Comment:
It takes me some time to figure out why it needs `once`. Please add comment why using `r.once` is needed.

Reply:
added comments.